/*

   calin/proto/iact_data/cta_data_source.proto -- Stephen Fegan -- 2018-11-06

   Protobufs for configuration of CTAZFITSDataSource

   Copyright 2018, Stephen Fegan <sfegan@llr.in2p3.fr>
   Laboratoire Leprince-Ringuet, CNRS/IN2P3, Ecole Polytechnique, Institut Polytechnique de Paris

   This file is part of "calin"

   "calin" is free software: you can redistribute it and/or modify it
   under the terms of the GNU General Public License version 2 or
   later, as published by the Free Software Foundation.

   "calin" is distributed in the hope that it will be useful, but
   WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   General Public License for more details.

*/

syntax = "proto3";

import "calin.proto";
import "iact_data/nectarcam_data_source.proto";
import "iact_data/lstcam_data_source.proto";

package calin.ix.iact_data.cta_data_source;

enum CameraType {
  AUTO_DETECT                                            = 0;
  NECTARCAM                                              = 1;
  LSTCAM                                                 = 2;
};

message CTACameraEventDecoderConfig {
  CameraType camera_type                                   = 1 [
    (CFO).desc = "Camera type." ];

  calin.ix.iact_data.nectarcam_data_source.NectarCamCameraEventDecoderConfig nectarcam = 10 [
    (CFO).desc = "NectarCam decoder options." ];

  calin.ix.iact_data.lstcam_data_source.LSTCamCameraEventDecoderConfig lstcam = 11 [
    (CFO).desc = "LSTCam decoder options." ];
};
